app/Http/Controllers/ProductController.php

```diff
@@ -11,5 +11,12 @@ class ProductController extends Controller
     public function listProduct($id=null){
         return $id?Product::find($id):Product::all();
     }
+
+    //Searching for Categories
+    public function searchCategory($id){
```

Comment:
why are you using DB queries?
why you are not using Eloquent query?

Reply:
Sir I did the change in second baranch